This patch is a hand-edited CVS patch covering only minor important cleanup changes.
--- debugedit.c
+++ debugedit.c
@@ -153,7 +153,7 @@
 	{
 	  if (data->d_buf
 	      && offset >= data->d_off
-	      && offset < data->d_off + data->d_size)
+	      && offset < data->d_off + (off_t)data->d_size)
 	    return (const char *) data->d_buf + (offset - data->d_off);
 	}
     }
@@ -468,8 +465,8 @@
 has_prefix (const char  *str,
 	    const char  *prefix)
 {
-  int str_len;
-  int prefix_len;
+  size_t str_len;
+  size_t prefix_len;
   
   str_len = strlen (str);
   prefix_len = strlen (prefix);
@@ -747,8 +747,8 @@
   for (i = 0; i < t->nattr; ++i)
     {
       uint_32 form = t->attr[i].form;
-      uint_32 len = 0;
-      int base_len, dest_len;
+      size_t len = 0;
+      size_t base_len, dest_len;
       
 
       while (1)
